document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt

```diff
@@ -148,61 +102,35 @@ class SmartDocumentsClient(
         )
     }
 
-    private fun toHttpClientErrorException(e: Throwable): HttpClientErrorException {
-        if (e is WebClientResponseException) {
-            val message = when (e.statusCode) {
-                HttpStatus.UNAUTHORIZED -> "The request has not been applied because it lacks valid authentication " +
-                        "credentials for the target resource. Response received from server:\n" + e.responseBodyAsString
-
-                HttpStatus.BAD_REQUEST -> "The server cannot or will not process the request due to something that is " +
-                        "perceived to be a client error (e.g., no valid template specified, user has no privileges for the template," +
-                        " malformed request syntax, invalid request message framing, or deceptive request routing)." +
-                        " Response received from server:\n" + e.responseBodyAsString
-
-                else -> e.responseBodyAsString
-            }
-            return HttpClientErrorException(e.statusCode, message)
-        } else {
-            return HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR, e.message ?: "An unknown error occurred")
-        }
-    }
-
     fun setProperties(smartDocumentsConnectorProperties: SmartDocumentsConnectorProperties) {
         this.smartDocumentsConnectorProperties = smartDocumentsConnectorProperties
     }
 
-    private fun pluginWebClient(pluginProperties: SmartDocumentsPropertiesDto): WebClient {
-        val basicAuthentication = ExchangeFilterFunctions.basicAuthentication(
-            pluginProperties.username,
-            pluginProperties.password
-        )
-
-        return smartDocumentsWebClientBuilder
+    private fun pluginRestClient(pluginProperties: SmartDocumentsPropertiesDto): RestClient {
+        return smartDocumentsRestClientBuilder
             .clone()
             .baseUrl(pluginProperties.url)
-            .filter(basicAuthentication)
-            .build()
-    }
-
-    private fun webClient(): WebClient {
-        val basicAuthentication = ExchangeFilterFunctions.basicAuthentication(
-            smartDocumentsConnectorProperties.username!!,
-            smartDocumentsConnectorProperties.password!!
-        )
-
-        // Setting the max file size for the smart documents response
-        val exchangeStrategies = ExchangeStrategies
-            .builder()
-            .codecs { configurer: ClientCodecConfigurer ->
-                configurer.defaultCodecs().maxInMemorySize(1024 * 1024 * maxFileSizeMb)
```

Comment:
We might need to find a replacement for this ^. Because the commit message here says:
```
bugfix Smart documents memory exceptions (#198)

Smart Documents could return a base64 string of a document that is bigger than the default 256 kb. Therefor the smart documents webClient will be configured for a larger memory size
```

Reply:
I found this class to handle Resource streaming content. After reading this post I feel this solves the issue. https://stackoverflow.com/questions/36379835/getting-inputstream-with-resttemplate
Would like to test it someway.... Any ideas on that?